test/basic.html

```diff
@@ -257,5 +249,78 @@
         });
       });
     });
+
+    describe('item components', () => {
+      let menu, buttons, ellipsis;
+
+      function makeComponent(id) {
+        const div = document.createElement('div');
+        div.style.width = '100px';
+        div.textContent = `Item ${id}`;
+        return div;
+      }
+
+      beforeEach(async() => {
+        menu = fixture('default');
+        menu.items = [
+          {text: 'Item 1'},
+          {text: 'Item 2'},
+          {component: makeComponent('3')},
+          {text: 'Item 4 text', component: makeComponent('4')},
+          {text: 'Item 5', component: document.createElement('vaadin-context-menu-item')}
+        ];
+        await nextRender(menu);
+        buttons = menu._buttons;
+        ellipsis = buttons[buttons.length - 1];
+      });
+
+      it('should render the component inside the context-menu item', () => {
+        const item = buttons[2].firstChild;
+        expect(item).to.equal(buttons[2].item.component);
+        expect(item.localName).to.equal('vaadin-context-menu-item');
+        const div = item.firstChild;
+        expect(div).to.equal(menu.items[2].component);
+        expect(div.localName).to.equal('div');
+        expect(div.textContent).to.equal('Item 3');
+        expect(getComputedStyle(div).width).to.equal('100px');
+      });
+
+      it('should override the component text when defined on the item', () => {
+        const item = buttons[3].firstChild;
+        expect(item).to.equal(buttons[3].item.component);
+        expect(item.localName).to.equal('vaadin-context-menu-item');
+        const div = item.firstChild;
+        expect(div).to.equal(menu.items[3].component);
+        expect(div.localName).to.equal('div');
+        expect(div.textContent).to.equal('Item 4 text');
+        expect(getComputedStyle(div).width).to.equal('100px');
+      });
+
+      it('should render provided context-menu item as a component', () => {
+        expect(buttons[4].item.component).to.equal(menu.items[4].component);
+        expect(buttons[4].firstChild).to.equal(menu.items[4].component);
+        expect(buttons[4].firstChild.textContent).to.equal('Item 5');
```

Comment:
`buttons[4].firstChild` in this case is `vaadin-context-menu-item` instance.

> Just to be safe, I think we should also verify that the item has no child elements.

Does that mean that `vaadin-context-menu-item` created by Flow can only contain text? What about providing custom components, like icons, inside it?

Asking as this feature only exists for Flow so we can tweak it according to our needs.

Reply:
My comment was actually not related to Flow. I'll try to explain my point better:
- the implemented logic creates automatically `vaadin-context-menu-items` under certain conditions
- in this test case, when the user provided an instance of `vaadin-context-menu-item`, the implemented logic does not add any "extra" element automatically
- but since it may do so in other conditions, we should test that nothing extra is added in this case.
- testing the `textContent` doesn't verify that no extra elements were added in-between

Reply:
I understand it now, thanks.

Reply:
Test updated.